<a href="https://colab.research.google.com/github/saleemshad/Full-Stack-AI-ML/blob/main/EDA_with_Pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Exploratory Data Analysis

## importing libraries

In [4]:
import numpy as np
import pandas as pd

## importing data from google drive to load the file in pandas dataframe

In [5]:
data_file = pd.read_csv('/content/drive/MyDrive/geolocations_amenities_ghmc.csv')

## lookig at the over picture of the dataset

In [6]:
data_file.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88718 entries, 0 to 88717
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Sno               88718 non-null  int64  
 1   Wardno Name       88713 non-null  object 
 2   Circleno Name     88714 non-null  object 
 3   Zone              88699 non-null  object 
 4   Assessor Address  88666 non-null  object 
 5   Latitude          88717 non-null  float64
 6   Longitude         88717 non-null  float64
 7   Category          88718 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 5.4+ MB


## descrivin the dataset more to get insights

In [7]:
data_file.describe()

Sno      Latitude     Longitude
count  88718.000000  88717.000000  88717.000000
mean   44359.500000     19.480063     76.413930
std    25610.824928     11.012548     11.005354
min        1.000000     17.135225     17.282086
25%    22180.250000     17.381477     78.417444
50%    44359.500000     17.425307     78.465952
75%    66538.750000     17.482789     78.525518
max    88718.000000     78.620648     78.700814

### Missing data
Since Latitude and Longitude are not null but their count of values is one less that is : 88717 out of 88718 which means it has one null values.
We can confirm this by ruuning the command below.
The results below show.
* Both Latitude and Longitude has a null values

In [8]:
data_file['Latitude'].isnull().values.any()

True

In [9]:
data_file['Longitude'].isnull().values.any()

True

In [10]:
data_file['Latitude'].isnull().sum()


1

In [11]:
data_file['Longitude'].isnull().sum()

1

### Checking the location of Nan
We can also see where extactly the Nan values in the Longitude

In [18]:
nan_values = data_file[data_file['Longitude'].isna()]
nan_values

Sno Wardno Name Circleno Name  ... Latitude Longitude          Category
88665  88666         NaN           NaN  ...      NaN       NaN  SPORTS COMPLEXES

[1 rows x 8 columns]

## Replacing Nan with 0
Now we are replacing NaN values with 0 in both Latitude and Longitude

In [19]:
data_file['Latitude'] = data_file['Latitude'].fillna(0)

In [20]:
data_file['Longitude'] = data_file['Longitude'].fillna(0)

## verify the after modification

We are now verifying again whether we have any nulls values
so now we get empty result that means the values were updated with 0

In [21]:
nan_values = data_file[data_file['Longitude'].isna()]
nan_values

Empty DataFrame
Columns: [Sno, Wardno Name, Circleno Name, Zone, Assessor Address, Latitude, Longitude, Category]
Index: []